config.py

```diff
@@ -1427,9 +1453,72 @@ def get_sessions():
         return sessions_
 
 
-def get_runs() -> list:
+def get_runs_all_subjects() -> dict:
+    '''Returns a dict of runs present in the bids_path
+    for each subject asked in the configuration file
+    (and not for each subject present in the bids_path).'''
+    # We cannot use get_subjects() because if there is just one subject
+    # we need to creat the full list of ignore_subjects
+    valid_subs = get_entity_vals(get_bids_root(), entity_key='subject')
+
+    subj_runs = dict()
+    for subj in get_subjects():
+        # ignore all subject but the one considered
+        # Not very elegant, but selecting one subject is not possible
+        # with the current api of get_entity_vals in mne-bids
+        ignore_subjects = valid_subs.copy()
+        if subj in ignore_subjects:
+            ignore_subjects.remove(subj)
+        else:
+            ValueError(f"{subj} not in {ignore_subjects}")
+
+        valid_runs_subj = get_entity_vals(
+            get_bids_root(), entity_key='run',
+            ignore_subjects=ignore_subjects)
+        subj_runs[subj] = valid_runs_subj
+
+    return subj_runs
+
+
+def get_intersect_run() -> list:
+    '''Returns the intersection of all the runs of all subjects.'''
+    subj_runs = get_runs_all_subjects()
+    return list(set.intersection(*map(set, subj_runs.values())))
+
+
+def get_runs(subject: str, verbose: bool = False) -> list:
+    """Returns a list of runs.
+
+    Parameters
+    ----------
+    subject
+        Returns a list of the runs of this subject.
+    verbose
+        Notify if different subjects do not share the same runs.
+
+    Returns
+    -------
+    list of string
+        list of runs of the subject.
+    """
     runs_ = copy.deepcopy(runs)  # Avoid clash with global variable.
-    valid_runs = get_entity_vals(get_bids_root(), entity_key='run')
+
+    subj_runs = get_runs_all_subjects()
+    valid_runs = subj_runs[subject]
+
+    if len(get_subjects()) > 1:
+        # Notify if different subjects do not share the same runs
+
+        same_runs = True
+        for runs_sub_i in subj_runs.values():
```

Comment:
maybe something like

`if len(set(subj_runs.values())) > 2:`

ie enough to know if some subjects have different number of runs?

Reply:
That was what I've first tried, but then I got the error : unhashable type : list

Reply:
then maybe:

`if len(set(map(tuple, subj_runs.values()))) > 2:`

?